megamek/src/megamek/common/weapons/BayWeaponHandler.java

```diff
@@ -498,8 +540,13 @@ public boolean handleAeroSanity(IGame.Phase phase, Vector<Report> vPhaseReport)
                 if(bayWType instanceof Weapon) {
                     replaceReport = vPhaseReport.size();
                     WeaponAttackAction bayWaa = new WeaponAttackAction(waa.getEntityId(), waa.getTargetType(), waa.getTargetId(), wId);
-                    AttackHandler bayWHandler = ((Weapon)bayWType).getCorrectHandler(toHit, bayWaa, game, server);
+                    AttackHandler bayWHandler = ((Weapon)bayWType).getCorrectHandler(autoHit, bayWaa, game, server);
                     bayWHandler.setAnnouncedEntityFiring(false);
+                    // This should always be true. Maybe there's a better way to write this?
+                    if (bayWHandler instanceof WeaponHandler) {
```

Comment:
as in `bayWHandler` is always a `WeaponHandler`?

Reply:
Yes

Reply:
Then I'd add in an else clause which raises a warning if it isn't.

Reply:
Fixed